src/core/optctrl/shooting.cpp

```diff
@@ -66,23 +68,36 @@ double ShootingProblem::calcDiff(const std::vector<Eigen::VectorXd>& xs, const s
                  << "us has wrong dimension (it should be " + std::to_string(T_) + ")");
   }
 
-  cost_ = 0;
   std::size_t i;
 
 #ifdef WITH_MULTITHREADING
   omp_set_num_threads(NUM_THREADS);
+#endif
+
+  if (recalc) {
```

Comment:
Remove the `recalc` option. Let's follow the same patter to this class too.
Move the `calc` parallelization to the `ShootingProblem::calc`.

Reply:
This recalc actually affects the code. And I need to think on how to do this properly, since this is the part being called in the solvers.
I'll handle it in a separate PR. I don't want to push this change with this big PR.